clang/lib/StaticAnalyzer/Checkers/WebKit/RawPtrRefMemberChecker.cpp

```diff
@@ -87,6 +92,31 @@ class RawPtrRefMemberChecker
     }
   }
 
+  void visitObjCDecl(const ObjCContainerDecl *CD) const {
+    if (auto *ID = dyn_cast<ObjCInterfaceDecl>(CD)) {
+      for (auto *Ivar : ID->ivars())
+        visitIvarDecl(CD, Ivar);
+      return;
+    }
+    if (auto *ID = dyn_cast<ObjCImplementationDecl>(CD)) {
+      for (auto *Ivar : ID->ivars())
+        visitIvarDecl(CD, Ivar);
+      return;
+    }
+  }
+
+  void visitIvarDecl(const ObjCContainerDecl *CD,
+                     const ObjCIvarDecl *Ivar) const {
+    const Type *IvarType = Ivar->getType().getTypePtrOrNull();
+    if (!IvarType)
+      return;
+    if (auto *IvarCXXRD = IvarType->getPointeeCXXRecordDecl()) {
```

Comment:
Can `IvarCXXRD` be null?

Reply:
I think so. If `IvarType` isn't a pointer to a C++ type, `getPointeeCXXRecordDecl` can return `nullptr`. e.g. Objective-C interface type.

Reply:
But again, we're doing a nullptr check here with if statement so I think we're good :)